docs_src/microservices/application/Triggers.md

```diff
@@ -27,13 +31,98 @@ Type = 'zero' #specifies of message bus (i.e zero for ZMQ)
+By default, `EdgeX Core Data` publishes data to the `events`  topic using ZMQ on port 5563. The publish host is used if publishing data back to the message bus. As stated above there are three implementations you can choose from. These type values are as follows:
+
+```
+zero - for ZeroMQ
+mqtt - for MQTT (Requires a MQTT Broker running and Core-Data configure to use MQTT)
+redisstreams - for Redis Streams (Requires Redis running and Core-Data configure to use Redis Streams)
+```
 
 !!! important
 When using ZMQ for the message bus, the Publish Host **MUST** be different for every topic you wish to publish to since the SDK will bind to the specific port. 5563 for example cannot be used to publish since `EdgeX Core Data` has bound to that port. Similarly, you cannot have two separate instances of the app functions SDK running publishing to the same port.
 
+!!! note
+When using MQTT for the message bus, there is additional configuration required for specifying the MQTT specific options. 
+
+Here is example `MessageBus` configuration when using MQTT as the message bus:
+
+```toml
+[MessageBus]
+Type = "mqtt"
+    [MessageBus.SubscribeHost]
+        Host = "localhost"
+        Port = 1883
+        Protocol = "tcp"
+    [MessageBus.PublishHost]
+        Host = "localhost"
+        Port = 1883
+        Protocol = "tcp"
+    [MessageBus.Optional]
+        # MQTT Specific options
+        Username =""
+        Password =""
+        ClientId ="AppService"
+        # Connection information
+        Qos          =  "0" # Quality of Sevice 0 (At most once), 1 (At least once) or 2 (Exactly once)
+        KeepAlive    =  "10" # Seconds (must be 2 or greater)
+        Retained     = "false"
+        AutoReconnect  = "true"
+        ConnectTimeout = "5" # Seconds
+        SkipCertVerify = "false"
+        CertFile       = ""
+        KeyFile        = ""
+        KeyPEMBlock    = ""
+        CertPEMBlock   = ""
+```
+
+!!! note
+The MQTT `MessageBus` implementation doesn't yet support retrieving secrets from the `Secret Store`. Thus the secrets for secure connection are currently in plain text in the configuration file.
+
+## MQTT Trigger
+
+A MQTT trigger will execute the pipeline every time data is received from the configured external MQTT broker on the configured subscribe topic.  
+
+!!! note
+The data received from the external MQTT broker is not wrapped with any metadata known to EdgeX. The data is handled as JSON or CBOR. The data is assumed to be JSON unless the first byte in the data is not a `{` , in which case it is then assumed to be CBOR.
+
+!!! note
+The data receive, encoded as JSON or CBOR, must match the `TargetType` define by your application service. The default  `TargetType` is an `Edgex Event`. See [TargetType](../AdvancedTopics/#target-type) for more details.
+
+### Type and Topic configuration 
+Here's an example:
+```toml
+Type="external-mqtt" 
+SubscribeTopic="some-request"
+PublishTopic=""
```

Comment:
Shall we consider to include `[Binding]` to make this configuration easier to understand?